### PR TITLE
Issue #9486: updated AST for TokenTypes.LITERAL_NEW

### LINKED ISSUE #9486: update example of AST for TokenTypes.LITERAL_NEW
https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_NEW

needs to be extended to show example of AST.
Source to be updated: https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java

PLEASE READ for more details: https://github.com/checkstyle/checkstyle/wiki/Good-practice-on-how-to-update-AST-in-TokenTypes

### PR DESCRIPTION
**Pull request by Rahulkhinchi03**
fixes #9486 

Format

Generated Java Doc

Final JavaDoc

> Code used for generating example:
<pre>
<code>
import java.util.ArrayList;
public class MyClass {
     public static void main(String[] args) {
        new ArrayList<>(50);
    }
}
</code>
</pre>

> Output in AST Format:
<pre>
<code>
IMPORT -> import [1:0]
|--DOT -> . [1:16]
|   |--DOT -> . [1:11]
|   |   |--IDENT -> java [1:7]
|   |   `--IDENT -> util [1:12]
|   `--IDENT -> ArrayList [1:17]
`--SEMI -> ; [1:26]
CLASS_DEF -> CLASS_DEF [2:0]
|--MODIFIERS -> MODIFIERS [2:0]
|   `--LITERAL_PUBLIC -> public [2:0]
|--LITERAL_CLASS -> class [2:7]
|--IDENT -> MyClass [2:13]
`--OBJBLOCK -> OBJBLOCK [2:21]
    |--LCURLY -> { [2:21]
    |--METHOD_DEF -> METHOD_DEF [3:5]
    |   |--MODIFIERS -> MODIFIERS [3:5]       
    |   |   |--LITERAL_PUBLIC -> public [3:5] 
    |   |   `--LITERAL_STATIC -> static [3:12]
    |   |--TYPE -> TYPE [3:19]
    |   |   `--LITERAL_VOID -> void [3:19]
    |   |--IDENT -> main [3:24]
    |   |--LPAREN -> ( [3:28]
    |   |--PARAMETERS -> PARAMETERS [3:35]
    |   |   `--PARAMETER_DEF -> PARAMETER_DEF [3:35]
    |   |       |--MODIFIERS -> MODIFIERS [3:35]
    |   |       |--TYPE -> TYPE [3:35]
    |   |       |   `--ARRAY_DECLARATOR -> [ [3:35]
    |   |       |       |--IDENT -> String [3:29]
    |   |       |       `--RBRACK -> ] [3:36]
    |   |       `--IDENT -> args [3:38]
    |   |--RPAREN -> ) [3:42]
    |   `--SLIST -> { [3:44]
    |       |--EXPR -> EXPR [4:8]
    |       |   `--LITERAL_NEW -> new [4:8]
    |       |       |--IDENT -> ArrayList [4:12]
    |       |       |--TYPE_ARGUMENTS -> TYPE_ARGUMENTS [4:21]
    |       |       |   |--GENERIC_START -> < [4:21]
    |       |       |   `--GENERIC_END -> > [4:22]
    |       |       |--LPAREN -> ( [4:23]
    |       |       |--ELIST -> ELIST [4:24]
    |       |       |   `--EXPR -> EXPR [4:24]
    |       |       |       `--NUM_INT -> 50 [4:24]
    |       |       `--RPAREN -> ) [4:26]
    |       |--SEMI -> ; [4:27]
    |       `--RCURLY -> } [5:4]
    `--RCURLY -> } [6:0]
</code>
</pre>

> Expected Java Doc to be Updated:

<pre>
<code>
     * LITERAL_NEW -> new
     *  |--IDENT -> ArrayList
     *  |--TYPE_ARGUMENTS -> TYPE_ARGUMENTS
     *  |   |--GENERIC_START -> <
     *  |   `--GENERIC_END -> >
     *  |--LPAREN -> (
     *  |--ELIST -> ELIST
     *  |       |   `--EXPR -> EXPR
     *  |       |       `--NUM_INT -> 50
     *  `--RPAREN -> )
    </code>
</pre>

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 1]
```diff
@@ -3874,18 +3874,18 @@
...
-     *     +--IDENT (ArrayList)
-     *     +--LPAREN (()
-     *     +--ELIST
-     *         |
-     *         +--EXPR
-     *             |
-     *             +--NUM_INT (50)
-     *     +--RPAREN ())
-     * </pre>
+     *       <pre>
```
Comment (Rahulkhinchi03) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:3877]: spaces should be deleted

Comment (AkMo3) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:3877]: `<pre>` should not be indented

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 2]
```diff
@@ -3874,18 +3874,18 @@
...
+     * LITERAL_NEW -&t; new
```
Comment (AkMo3) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:3878]: Change all occurring of `&t;` to `&gt;`

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 3]
```diff
@@ -3874,18 +3874,18 @@
...
+     *  |--IDENT -&t; ArrayList
+     *  |--TYPE_ARGUMENTS -&t; TYPE_ARGUMENTS
+     *  |   |--GENERIC_START -&t;
+     *  |   `--GENERIC_END -&t;
+     *  |--LPAREN -&t; (
+     *  |--ELIST -&t; ELIST
+     *  |       |   `--EXPR -&t; EXPR
+     *  |       |       `--NUM_INT -&t; 50
```
Comment (AkMo3) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:3886]: Change to 
```          
     *  |--ELIST -&gt; ELIST
     *  |   `--EXPR -&gt; EXPR
     *  |        `--NUM_INT -&gt; 50
```

**Review by AkMo3 — CHANGES_REQUESTED** [review]
@Rahulkhinchi03
Change commit message to `Issue #9486: updated AST for TokenTypes.LITERAL_NEW` , observe there is no space between colon and issue number

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java (hunk 1 above)
Comment (Rahulkhinchi03) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:3877]: How can I change my mistakes @AkMo3

Comment (AkMo3) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:3877]: 1. Change the file. 
2. Run `mvn clean verify`
3. Commit new changes. 
4. Squash your commit into one, see [this](https://davidwalsh.name/squash-commits-git) if you require help.
5. Force push changes

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java (hunk 2 above)
Comment (Rahulkhinchi03) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java]: Resolved this

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java (hunk 3 above)
Comment (Rahulkhinchi03) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java]: Resolved this.

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java (hunk 2 above)
Comment (AkMo3) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java]: Resolved, please click resolve from your side.

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java (hunk 3 above)
Comment (AkMo3) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java]: Resolved

File: pom.xml [hunk 4]
```diff
@@ -381,6 +381,11 @@
         </exclusion>
       </exclusions>
     </dependency>
+      <dependency>
+          <groupId>com.puppycrawl.tools</groupId>
+          <artifactId>checkstyle</artifactId>
+          <version>8.40</version>
+      </dependency>
```
Comment (AkMo3) [inline pom.xml:388]: Remove these changes

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 5]
```diff
@@ -3874,18 +3874,18 @@
...
-     *             |
-     *             +--NUM_INT (50)
-     *     +--RPAREN ())
-     * </pre>
+     *       <pre>
+     * LITERAL_NEW -&gt; new
+     *  |--IDENT -&gt; ArrayList
+     *  |--TYPE_ARGUMENTS -&gt; TYPE_ARGUMENTS
+     *  |   |--GENERIC_START -&gt;
+     *  |   `--GENERIC_END -&gt;
```
Comment (AkMo3) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:3882]:
```
     *  |   |--GENERIC_START -&gt; &lt;
     *  |   `--GENERIC_END -&gt; &gt;
```

**Review by AkMo3 — CHANGES_REQUESTED** [review]
Few more: 
Please squash commits into one

**Review by Rahulkhinchi03 — COMMENTED** [review]
done Changes and Reviewed.

File: pom.xml (hunk 4 above)
Comment (AkMo3) [reply pom.xml]: Not resolved

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java (hunk 1 above)
Comment (AkMo3) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:3877]: Not resolved

**Review by AkMo3 — COMMENTED** [review]
I think you were not able to squash commits properly so previous 2 commits also squashed up here. A lot of unnecessary file needs to be removed. Remove `git push` file, and fix commit structure.

**Comment by Rahulkhinchi03** [comment]
@AkMo3  please check. only 1 is failed. How can I improve it?

**Comment by AkMo3** [comment]
Your changes are no longer present. Please check changed files.

**Pull request closed without merging**